src/react-components/banner/Banner.js

```diff
@@ -0,0 +1,321 @@
+import React, { useState, useCallback } from "react";
+import { RadioInputField } from "../input/RadioInputField";
+import { RadioInputOption } from "../input/RadioInput";
+import { Button } from "../input/Button";
+import { CloseButton } from "../input/CloseButton";
+import { useForm } from "react-hook-form";
+import styles from "./Banner.scss";
+import { ReactComponent as ArrowIcon } from "../icons/Arrow.svg";
+import warning_icon from "../../assets/images/warning_icon.png";
+import { TextInputField } from "../input/TextInputField";
+import { CheckboxInput } from "../input/CheckboxInput";
+import { FormattedMessage } from "react-intl";
+import cone from "../../assets/images/cone.png";
+
+const Banner = () => {
+  const [email, setEmail] = useState("");
+  const [confirm, setConfirm] = useState(true);
+  const [submitted, setSubmitted] = useState(false);
+  const [responseStatus, setResponseStatus] = useState(false);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const { handleSubmit, register } = useForm({
+    defaultValues: {
+      email_format: "html"
+    }
+  });
+
+  /**
+   * On Form Error
+   */
+  const newsletterError = () => {
+    setResponseStatus(false);
+    setSubmitted(true);
+  };
+
+  /**
+   * On Form Success
+   */
+  const newsletterSuccess = () => {
+    setResponseStatus(true);
+    setSubmitted(true);
+  };
+
+  /**
+   * On XHR Load
+   * @param {Basket Response} resp
+   * @returns
+   */
+  const onload = resp => {
+    // Check Target Status
+    const status = resp.target.status;
+    if (status !== 200 || status > 300 || status < 200) {
+      newsletterError();
+      return;
+    }
+
+    // Check Response Status
+    let response = resp.target.response;
+    typeof response !== "object" ? (response = JSON.parse(response)) : "";
+    if (response.status !== "ok") {
+      newsletterError();
+      return;
+    }
+
+    // Do success stuff here..
+    newsletterSuccess();
+  };
+
+  const onSubmit = async data => {
+    // Note: this is a pattern that was suggesed by the basket creator
+    const url = "https://basket.mozilla.org/news/subscribe/";
+    const xhr = new XMLHttpRequest();
+    const params =
+      "email=" +
+      encodeURIComponent(data.email) +
+      "&newsletters=" +
+      "hubs" +
+      "&lang=" +
+      encodeURIComponent(navigator.languages[0]) +
```

Comment:
`navigator.language` is simpler and available more browser versions.
```suggestion
      encodeURIComponent(navigator.language) +
```